src/main/java/org/apache/maven/plugins/checkstyle/rss/VelocityTemplate.java

```diff
@@ -77,52 +79,34 @@ public VelocityComponent getVelocity()
      * Using a specified Velocity Template and provided context, create the outputFilename.
      *
      * @param outputFilename the file to be generated.
-     * @param template       the velocity template to use.
-     * @param context        the velocity context map.
-     * @throws VelocityException if the template was not found or any other Velocity exception.
-     * @throws MojoExecutionException if merging the velocity template failed.
-     * @throws IOException if there was an error when writing to the output file.
+     * @param template       the velocity template to use
+     * @param context        the velocity context map
+     * @throws VelocityException if the template was not found or any other Velocity exception
+     * @throws MojoExecutionException if merging the velocity template failed
+     * @throws IOException if there was an error writing to the output file
      */
     public void generate( String outputFilename, String template, Context context )
         throws VelocityException, MojoExecutionException, IOException
     {
-        Writer writer = null;
 
-        try
+        File f = new File( outputFilename );
+        if ( !f.getParentFile().exists() )
+        {
+            f.getParentFile().mkdirs();
+        }
+        
+        try ( Writer writer = new OutputStreamWriter( new FileOutputStream( f ), StandardCharsets.UTF_8 ) )
         {
```

Comment:
Can you pick a more descriptive variable name for 'f' here? We have `outputFilename` already, so `outputFile` would be an obvious choice.

Reply:
done